application_test.go

```diff
@@ -450,17 +450,60 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
 
-func TestWaitOnApplication(t *testing.T) {
+func TestWaitOnApplicationWithTimeoutGreaterThanTicker(t *testing.T) {
+	defaultConfig := NewDefaultConfig()
+	configs := &configContainer{}
+	configs.client = &defaultConfig
+	configs.client.WaitTime = 100
+
+	endpoint := newFakeMarathonEndpoint(t, configs)
+	defer endpoint.Close()
+
+	var err error
+	go func() {
+		err = endpoint.Client.WaitOnApplication(fakeAppName, 200*time.Millisecond)
+	}()
+	timer := time.NewTimer(500*time.Millisecond)
+	<- timer.C
+	assert.NoError(t, err)
+
+
+	err = nil
+	go func() {
+		err = endpoint.Client.WaitOnApplication("no_such_app", 200*time.Millisecond)
+	}()
+	timer = time.NewTimer(500*time.Millisecond)
+	<- timer.C
+	assert.IsType(t, err, ErrTimeoutError)
+}
+
+func TestWaitOnApplicationWithTimeoutSmallerThanTicker(t *testing.T) {
+	defaultConfig := NewDefaultConfig()
+	configs := &configContainer{}
+	configs.client = &defaultConfig
+	configs.client.WaitTime = 200
+
 	endpoint := newFakeMarathonEndpoint(t, nil)
 	defer endpoint.Close()
 
 	var err error
-	err = endpoint.Client.WaitOnApplication(fakeAppName, 1*time.Second)
+	go func() {
+		err = endpoint.Client.WaitOnApplication(fakeAppName, 100*time.Millisecond)
+	}()
+	timer := time.NewTimer(500*time.Millisecond)
+	<- timer.C
 	assert.NoError(t, err)
 
-	err = endpoint.Client.WaitOnApplication("no_such_app", 1*time.Millisecond)
+
+	err = nil
```

Comment:
I think we should play safe here and use a new error variable to rule out interferences between the previous invocation of `WaitOnApplication` (line 491) and the following one (line 500) in case of a broken implementation.

However, see also my comment on converting both new test functions into a single, table-driven test.